models.py

```diff
@@ -103,4 +127,24 @@ def calculate_centroids(self, support, num_classes): #, query_labels#, train_ite
 
         return centroids
 
-
+    def save_model(self, unfreeze_num, snapshot_path):
```

Comment:
Looking good, nice to have this in. On thing though, what do you think about not passing `unfreeze_num` for both for save and load? Since we are saving a dictionary structure we could get the number of unfreeze layers from the model itself and store it in a new key when saving. Then read that key when loading it. What do you think?

Reply:
yeah I guess that would be nicer in general but would also entail extracting the number from the dictionary key strings which is a bit messy. I'll add it later if I have time :)

Reply:
I mean as a new key-value pair, so when saving we'll do something like:
```python
state_dicts['unfreeze_num'] = model. unfreeze_num
```
And when loading
```python
checkpoint = torch.load(path, map_location=device)
unfreeze_num = checkpoint['unfreeze_num']
...
```

Reply:
ah yes that is a good idea. I didn't think of that. I'll fix it quickly

Reply:
Kk I fixed it :)